File: daehwa_park/12주차/PGS_연속_부분_수열_합의_ 개수.py
def solution(elements):
    answer = 0
    
    dic = {}
    
    for i in range(1, len(elements)):
        s = 0
        e = i
        snum = sum(elements[:i])
        dic.setdefault(snum, 0)
        for j in range(len(elements) - 1):
            snum -= elements[s]
            snum += elements[e]
            dic.setdefault(snum, 0)
            s += 1
            e = (e + 1) % len(elements)
    answer = len(dic) + 1
    return answer

File: sylee723/16주차/PGS_연속_부분_수열_합의_개수.java
import java.util.*;

class Solution {
    public int solution(int[] elements) {
        Set<Integer> set = new HashSet<>();
        
        int len = elements.length;
        for (int i = 0; i < len; i++) {
            int sum = 0;
            for (int l = 0; l < len; l++) {
                int idx = (i + l) % len; 
                sum += elements[idx];
                
                set.add(sum);
            }
        }
        
        int answer = set.size();
        return answer;
    }
}